File: EventsDetailFragment/app/src/main/java/com/example/eventsdetailfragment/Fragments/TimeAndDateFragment.java
package com.example.eventsdetailfragment.Fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.eventsdetailfragment.FragmentCommunication;
import com.example.eventsdetailfragment.R;

public class TimeAndDateFragment extends Fragment {

    private FragmentCommunication communication;
    private Button btnNextDate;
    private EditText etStartDate, etEndDate, etStartTime, etEndTime;
    private String eventTitle;
    private String eventDesc;
    private FragmentCommunication fragmentCommunication;


    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        communication = (FragmentCommunication) context;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            eventTitle = getArguments().getString("title");
            eventDesc = getArguments().getString("description");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_time_and_date, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
    }

    private void initViews(View view) {
        btnNextDate = view.findViewById(R.id.btnNextDate);
        etStartDate = view.findViewById(R.id.etStartDate);
        etEndDate = view.findViewById(R.id.etEndDate);
        etStartTime = view.findViewById(R.id.etStartTime);
        etEndTime = view.findViewById(R.id.etEndTime);

        btnNextDate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();

                String startDate = etStartDate.getText().toString();
                String endDate = etEndDate.getText().toString();
                String startTime = etStartTime.getText().toString();
                String endTime = etEndTime.getText().toString();
                eventTitle = getArguments().getString("title");
                eventDesc = getArguments().getString("description");
                bundle.putString("eventTitle", eventTitle);
                bundle.putString("eventDesc", eventDesc);
                bundle.putString("startDate", startDate);
                bundle.putString("endDate", endDate);
                bundle.putString("startTime", startTime);
                bundle.putString("endTime", endTime);

//                Toast.makeText(getContext(), "Data:" + eventTitle + "Desc " + eventDesc, Toast.LENGTH_SHORT).show();
                if (communication != null) {
                    communication.launchThirdFrag(bundle);
                }
            }
        });

    }
}

File: Kotlin Projects/EventsNavFragments/app/src/main/java/com/example/events_navfragments/TimeAndDateFragment.kt
package com.example.events_navfragments

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.Navigation
import kotlinx.android.synthetic.main.fragment_time_and_date.*

class TimeAndDateFragment : Fragment(R.layout.fragment_time_and_date) {

    private lateinit var navController: NavController
    private lateinit var eventName: String
    private lateinit var eventDecs: String


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getEventDetailsData()
    }

    private fun getEventDetailsData() {
        arguments?.run {
            eventName = getString("eventName").toString()
            eventDecs = getString("eventDecs").toString()
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        navController = Navigation.findNavController(view)

        btnNextTimeDate.setOnClickListener {
            val startTime = etStartTime.text.toString()
            val endTime = etEndTime.text.toString()
            val startDate = etStartDate.text.toString()
            val endDate = etEndDate.text.toString()

            val bundle = Bundle()
            bundle.putString("startTime", startTime)
            bundle.putString("endTime", endTime)
            bundle.putString("startDate", startDate)
            bundle.putString("endDate", endDate)
            bundle.putString("eventName", eventName)
            bundle.putString("eventDecs", eventDecs)
            navController.navigate(R.id.action_timeAndDateFragment_to_priceDetailsFragment, bundle)

        }
    }
}